release/stage-release.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -euo pipefail
+
+if ! command -v releaser >/dev/null; then
+    echo "releaser is not installed, install it by running: go install github.com/elastisys/releaser/cmd/releaser@latest" >&2
+    echo "For more information see https://github.com/elastisys/releaser/#installation" >&2
+    exit 1
+fi
+
+function usage() {
+    echo "Usage: ${0} VERSION" >&2
+    exit 1
+}
+
+[ ${#} -eq 1 ] || usage
+
+full_version="${1}"
+minor_version="$(echo "${full_version}" | cut -d '.' -f 1,2)"
+patch_version="$(echo "${full_version}" | cut -d '.' -f 3)"
+
+#
+# Create staging branch
+#
+
+git switch "release-${minor_version}"
+git pull
+git switch -c "staging-${full_version}"
+
+#
+# Cherry pick commits
+#
+
+for sha in ${CK8S_GIT_CHERRY_PICK:-}; do
+    git cherry-pick "${sha}"
+done
+
+#
+# Generate changelog
+#
+
+here="$(dirname "$(readlink -f "$0")")"
+changelog_dir="${here}/../changelog"
+changelog_path="${changelog_dir}/${minor_version}.md"
+
+mkdir -p "${changelog_dir}"
+
+# If this is a patch release, add an extra newline before appending the patch
+# notes. Also add an extra hashtag to please the markdownlint rule:
+# MD025 Multiple top level headers in the same document
+# TODO: Find a nicer way to do this.
+[ "${patch_version}" != "0" ] && printf "\n#" >> "${changelog_path}"
```

Comment:
So this increases the header level of the output from `releaser` ? What about other headers, won't they still be off by one? Couldn't that be done in `releaser` given that it also receives the full version number as argument?

Other possibility: if patch version ≠ 0 then `releaser | sed 's/^#/##/' >> changelog`

Reply:
Or perhaps a distinct top level title? i.e.

```markdown
# Changelog for Apps 0.33.x

## 0.33.0
...
## 0.33.1
...
```

Reply:
> So this increases the header level of the output from `releaser` ? What about other headers, won't they still be off by one? Couldn't that be done in `releaser` given that it also receives the full version number as argument?

Yea, it's a bit unfortunate but it does look ok-ish: https://github.com/elastisys/compliantkubernetes-redis/blob/4eb00edc38804ec5f7ecd128ef118ea2581b4b85/changelog/6.2.6.md

> Other possibility: if patch version ≠ 0 then `releaser | sed 's/^#/##/' >> changelog`

I actively chose not to do that to not make it look like patch release notes were less important than those mentioned at the top of the minor version release.

> Or perhaps a distinct top level title? i.e.
> 
> ```
> # Changelog for Apps 0.33.x
> 
> ## 0.33.0
> ...
> ## 0.33.1
> ...
> ```

I like this idea! I'm considering making [releaser](https://github.com/elastisys/releaser) even more specialized for our own workflow. For example, providing it a file to work against could make it render the changelog differently than if it just outputs to stdout. Then it would make sense to add that top-level header you suggest and increase the levels of the rest of the headers. https://github.com/elastisys/releaser/issues/12